### PR TITLE
fix(VSelect): change onMouseUp to handle menu toggle on mouse events

### LINKED ISSUE #6564: [Bug Report] Autocomplete menu icon doesn't close menu
### Versions and Environment
**Vuetify:** 1.5.2
**Last working version:** 1.0.19
**Vue:** 2.6.6
**Browsers:** Chrome 72.0.3626.109
**OS:** Windows 10

### Steps to reproduce
type something into the autocomplete to populate the dropdown list. try to click the menu icon on the right of the control to close the menu. be sure that the browser is tall enough so that the menu does not cover up the menu icon

### Expected Behavior
menu icon clicks should open and close the menu

### Actual Behavior
Menu doesnt close

### Reproduction Link
<a href="https://codepen.io/anon/pen/omRZba?&editable=true&editors=101" target="_blank">https://codepen.io/anon/pen/omRZba?&editable=true&editors=101</a>

### PR DESCRIPTION
**Pull request by webstp**
## Description
When append-inner was clicked, the menu could be opened but not closed, this was cause by onClick
always setting the isMenuActive to true. Made a change to only allow onMouseUp totoggle isMenuActive
state when the append-inner icon is clicked

## Motivation and Context
fix #6564

## How Has This Been Tested?
I create tests for the bug and ran the unit tests

## Markup:

<details>

```vue
<template>
  <v-autocomplete  :items="states"></v-autocomplete>
</template>

<script>
export default {
  data () {
    return {
      states: [
          'Alabama', 'Alaska', 'American Samoa', 'Arizona',
          'Arkansas', 'California', 'Colorado', 'Connecticut',
          'Delaware', 'District of Columbia', 'Federated States of Micronesia',
          'Florida', 'Georgia', 'Guam', 'Hawaii', 'Idaho',
          'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky',
          'Louisiana', 'Maine', 'Marshall Islands', 'Maryland',
          'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi',
          'Missouri', 'Montana', 'Nebraska', 'Nevada',
          'New Hampshire', 'New Jersey', 'New Mexico', 'New York',
          'North Carolina', 'North Dakota', 'Northern Mariana Islands', 'Ohio',
          'Oklahoma', 'Oregon', 'Palau', 'Pennsylvania', 'Puerto Rico',
          'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee',
          'Texas', 'Utah', 'Vermont', 'Virgin Island', 'Virginia',
          'Washington', 'West Virginia', 'Wisconsin', 'Wyoming',
        ]
    }
  }
}
</script>
```

</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [ ] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by nekosaur** [comment]
Please add Playground markup that shows the issue

**Comment by webstp** [comment]
Updated with Markup

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 1]
```diff
@@ -550,10 +550,16 @@ export default baseMixins.extend<options>().extend({
       }
       this.selectedIndex = -1
     },
-    onClick () {
+    onClick (e: MouseEvent) {
       if (this.isDisabled) return
+      const appendInner = this.$refs['append-inner']
 
-      this.isMenuActive = true
+      if (!(appendInner &&
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.ts]: Is it possible to make this more concise? It is hard to follow.

**Comment by johnleider** [comment]
I'd like to get this in for v2.2. If you don't have time let me know.

File: packages/vuetify/src/components/VSelect/VSelect.ts (hunk 1 above)
Comment (webstp) [reply packages/vuetify/src/components/VSelect/VSelect.ts]: I am unsure how to make this more concise, Would an added comment suffice? This is the same logic used at line 668 just flipped since we do not want to reopen the menu after onMouseUp would close it.

On thought I had would be to move this logic into its own function

```typescript
isAppendInner(e: MouseEvent) {
    // return true if append inner is present
    // and the target is itself or inside
    const appendInner = this.$refs['append-inner']
    return appendInner && (appendInner === e.target || (appendInner as { [key: string]: any }).contains(e.target))
}
```
then here it would look like 
```javascript
if (!this.isAppendInner(e)) {
```

Comment (johnleider) [reply packages/vuetify/src/components/VSelect/VSelect.ts]: Let's go with that option. Just make sure to have a blank line between the variable declaration and the return statement.

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 2]
```diff
@@ -550,10 +550,16 @@ export default baseMixins.extend<options>().extend({
       }
       this.selectedIndex = -1
     },
-    onClick () {
+    onClick (e: MouseEvent) {
       if (this.isDisabled) return
+      const appendInner = this.$refs['append-inner']
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSelect/VSelect.ts]: Add to the options interface on line 45 and you don't need the cast

**Comment by johnleider** [comment]
Requesting final comments @vuetifyjs/core-contributors @vuetifyjs/core-team
